internal/transformations/desugarer.go

```diff
@@ -579,7 +571,18 @@ func desugar(astPtr *ast.Node, objLevel int) (err error) {
 	return nil
 }
 
-func desugarFile(ast *ast.Node) error {
+// Desugar Jsonnet expressions to reduce the number of constructs the rest of the implementation
+// needs to understand.
+//
+// Note that despite the name, desugar() is not idempotent.  String literals have their escape
```

Comment:
It will unescape it again. So `"\"blah\""` becomes `"blah"` after first unescaping and if we tried to unescape it again we would get `blah`. This is not very clean and perhaps we should just add a separate field for that. 

Anyway for performance reasons it's better not to run desugaring on the same nodes again and again.